File: 2025-2/sesion-de-clase-11/SoftProgNegocio/BOImpl/AreaBOImpl.cs
using System;
using System.Collections.Generic;
using PUCP.SoftProg.Modelo;
using PUCP.SoftProg.Modelo.RRHH;
using PUCP.SoftProg.Negocio.BO;
using PUCP.SoftProg.Persistencia.DAO.RRHH;
using PUCP.SoftProg.Persistencia.DAOImpl.RRHH;

namespace PUCP.SoftProg.Negocio.BOImpl {
    public class AreaBOImpl : IAreaBO {
        private readonly IAreaDAO areaDAO;

        public AreaBOImpl() { 
            this.areaDAO = new AreaDAOImpl();
        }

        public void Eliminar(int id) {
            if (!this.areaDAO.Eliminar(id)) {
                throw new Exception("No se pudo eliminar el area");
            }
        }

        public void Guardar(Area area, Estado estado) {
            if (estado == Estado.Nuevo) {
                this.areaDAO.Crear(area);
            }
            else if (estado == Estado.Modificado) { 
                this.areaDAO.Crear(area);
            }
        }

        public List<Area> Listar() {
            return this.areaDAO.LeerTodos();
        }

        public Area Obtener(int id) {
            return this.areaDAO.Leer(id);
        }
    }
}

File: 2025-2/sesion-de-clase-21/java/SoftProg/SoftProgNegocio/src/main/java/pe/edu/pucp/inf30/softprog/boimpl/rrhh/AreaBOImpl.java
package pe.edu.pucp.inf30.softprog.boimpl.rrhh;

import java.util.List;
import pe.edu.pucp.inf30.softprog.bo.rrhh.AreaBO;
import pe.edu.pucp.inf30.softprog.dao.rrhh.AreaDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.rrhh.AreaDAOImpl;
import pe.edu.pucp.inf30.softprog.modelo.Estado;
import pe.edu.pucp.inf30.softprog.modelo.rrhh.Area;

/**
 *
 * @author eric
 */
public class AreaBOImpl implements AreaBO {
    private final AreaDAO areaDao;
    
    public AreaBOImpl() {
        this.areaDao = new AreaDAOImpl();
    }
    
    @Override
    public List<Area> listar() {
        return this.areaDao.leerTodos();
    }

    @Override
    public Area obtener(int id) {
        return this.areaDao.leer(id);
    }

    @Override
    public void eliminar(int id) {
        this.areaDao.eliminar(id);
    }

    @Override
    public void guardar(Area modelo, Estado estado) {
        if (estado == Estado.Nuevo) {
            int id = this.areaDao.crear(modelo);
            modelo.setId(id);
        }
        else {
            this.areaDao.actualizar(modelo);
        }
    }   
}
